server/service/mongo/kv/kv_dao.go

```diff
@@ -108,12 +110,23 @@ func updateKeyValue(ctx context.Context, kv *model.KVDoc) error {
 
 }
 
+//Extract key values
+func getValue(str string) string {
+	rex :=regexp.MustCompile(`\(([^)]+)\)`)
+	res :=rex.FindStringSubmatch(str)
+	return res[len(res)-1]
+}
+
 func findKV(ctx context.Context, domain string, project string, opts service.FindOptions) (*mongo.Cursor, int, error) {
 	collection := session.GetDB().Collection(session.CollectionKV)
 	ctx, _ = context.WithTimeout(ctx, opts.Timeout)
 	filter := bson.M{"domain": domain, "project": project}
 	if opts.Key != "" {
-		filter["key"] = opts.Key
+		if strings.HasPrefix(opts.Key, "beginWith"){
```

Comment:
consider about the open-close principle https://blog.csdn.net/u011250882/article/details/47358519

there are many article about OCP, please learn